### ISSUE TITLE
Add support for Adding Playlists and songs to MediaBrowser

### ISSUE THREAD

**Issue by stboch** [KoljaWindeler/ytube_music_player#14]
Similar to Spotify App,
Add Contents of Playlists to MediaBrowser HA component...

*** things of note ***
Existing known issue would be the length the URL is vaild for and needing to refresh the content if we can't get the url when a song is requested.
Limits on enumerating all playlists and songs in those playlists. (Other people have described issues with this) These actions would also further the need to ensure these tasks are using asyncio.)

**Comment by KoljaWindeler** [comment]
so, I've added the first implimentation
https://github.com/KoljaWindeler/ytube_music_player/blob/main/custom_components/ytube_music_player/browse_media.py
In general it shows the user playlists and user albums .. more can be added fairly easy.
I can see thumbnails and go "into" the playlist / Album, but nothing happens when i hit the "play" icon 

Any idea what's suppose to happen?

**Comment by KoljaWindeler** [comment]
Nevermind found it.

**Comment by KoljaWindeler** [comment]
first version (0.2) added as pre-release with media_browse support

**Comment by stboch** [comment]
Hey thanks for getting to this before I could... been busy with work...

https://github.com/home-assistant/core/homeassistant/components/netatmo/media_source.py

This is a good app with example of the media_source implementation... the important part is the async_resolve_media function that returns the url and Mime_type for the media players to use. 

If you get this working theoretically you won't need the local input lookups because the media browser items will be available to all the players in the browser ui. Also could player multiple songs on multiple devices. Risk though Playlist might not work because Google doesn't like songs playing on multiple devices.

**Comment by blair287** [comment]
> Hey thanks for getting to this before I could... been busy with work...
> 
> https://github.com/home-assistant/core/homeassistant/components/netatmo/media_source.py
> 
> This is a good app with example of the media_source implementation... the important part is the async_resolve_media function that returns the url and Mime_type for the media players to use.
> 
> If you get this working theoretically you won't need the local input lookups because the media browser items will be available to all the players in the browser ui. Also could player multiple songs on multiple devices. Risk though Playlist might not work because Google doesn't like songs playing on multiple devices.

Did this get added my media browser is empty in HA and this is closed with no info?

**Comment by KoljaWindeler** [comment]
I guess that we're talking about two different things here ... 
the ticket was about the support of the mediabrowser in general .. that was added some time ago.

My understanding is that you don't have any entries in your media_browser, correct? I saw that "confusing docu" item and added a guide to the readme https://github.com/KoljaWindeler/ytube_music_player/blob/main/README.md#step---by---step-guide
does that help?
